packages/api/src/routes/pagination.ts

```diff
@@ -0,0 +1,141 @@
+import { numericValue } from "@metriport/core/external/carequality/ihe-gateway-v2/schema";
+import { Config } from "@metriport/core/util/config";
+import { PaginatedResponse, ResponseMeta } from "@metriport/shared";
+import { Request } from "express";
+import { z } from "zod";
+import {
+  Pagination,
+  PaginationFromItem,
+  PaginationItem,
+  PaginationToItem,
+} from "../command/pagination";
+
+export const defaultItemsPerPage = 50;
+export const maxItemsPerPage = 500;
+
+export const queryMetaSchema = z.intersection(
+  z.union(
+    [
+      z.object({
+        fromItem: z.string().optional(),
+        toItem: z.never().optional(),
+      }),
+      z.object({
+        fromItem: z.never().optional(),
+        toItem: z.string().optional(),
+      }),
+    ],
+    { errorMap: () => ({ message: "Either fromItem or toItem can be provided, but not both" }) }
+  ),
+  z.object({
+    count: numericValue.optional().refine(count => (count ? count <= maxItemsPerPage : true), {
```

Comment:
if you reverse optional and refine you may not need the undefined check.

Maybe we check it's greater than 0? Count -1 wouldn't make sense.

Reply:
good catch, done